ath79: add support for MikroTik RouterBOARD wAP-2nD (wAP)

0x444D opened the pull request:
The MikroTik RouterBOARD wAP-2nd (sold as wAP) is a small
2.4 GHz 802.11b/g/n PoE-capable AP.

Specifications:
 - SoC: Qualcomm Atheros QCA9533
 - Flash: 16 MB (SPI)
 - RAM: 64 MB
 - Ethernet: 1x 10/100 Mbps (PoE in)
 - WiFi: AR9531 2T2R 2.4 GHz (SoC)
 - 3x green LEDs (1x lan, 1x wlan, 1x user)

 See https://mikrotik.com/product/RBwAP2nD for more info.

Flashing:
 TFTP boot initramfs image and then perform sysupgrade. Follow common
 MikroTik procedure as in https://openwrt.org/toh/mikrotik/common.

Note: following 781d4bfb397cdd12ee0151eb66c577f470e3377d
 The network setup avoids using the integrated switch and connects the
 single Ethernet port directly. This way, link speed (10/100 Mbps) is
 properly reported by eth0.

Signed-off-by: David Musil <0x444d@protonmail.com>

f00b4r0 commented on target/linux/ath79/mikrotik/base-files/etc/board.d/02_network:
```diff
@@ -20,7 +20,8 @@ ath79_setup_interfaces()
 	mikrotik,routerboard-sxt-5nd-r2|\
 	mikrotik,routerboard-wap-g-5hact2hnd|\
-	mikrotik,routerboard-wapr-2nd)
+	mikrotik,routerboard-wapr-2nd|\
+	mikrotik,routerboard-wap-2nd)
```
Alphabetical order please (wap-2nd before wap-g-5hact2hnd)

f00b4r0 commented on target/linux/ath79/mikrotik/base-files/etc/board.d/02_network:
```diff
@@ -44,7 +45,8 @@ ath79_setup_macs()
 	mikrotik,routerboard-mapl-2nd|\
 	mikrotik,routerboard-sxt-5nd-r2|\
 	mikrotik,routerboard-wap-g-5hact2hnd|\
-	mikrotik,routerboard-wapr-2nd)
+	mikrotik,routerboard-wapr-2nd|\
+	mikrotik,routerboard-wap-2nd)
```
same

f00b4r0 commented on target/linux/ath79/mikrotik/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:
```diff
@@ -27,7 +27,8 @@ case "$FIRMWARE" in
 	mikrotik,routerboard-lhg-5nd|\
 	mikrotik,routerboard-sxt-5nd-r2|\
-	mikrotik,routerboard-wapr-2nd)
+	mikrotik,routerboard-wapr-2nd|\
+	mikrotik,routerboard-wap-2nd)
```
wap-2nd before wapr-2nd

f00b4r0 commented on target/linux/ath79/dts/qca9533_mikrotik_routerboard-wap-2nd.dts:
```diff
@@ -0,0 +1,61 @@
+};
+
+
+&pinmux {
+	led_lan_pin: pinmux_led_lan_pin {
+		pinctrl-single,bits = <0x4 0x0 0xff>;
+	};
+};
```
Is this necessary? Looking at the [ar71xx code](https://github.com/openwrt/openwrt/blob/698cdf02020aa9855ab1000d6a359f816c76280e/target/linux/ar71xx/files/arch/mips/ath79/mach-rbspi.c#L1027), we didn't touch the pinmux on this device.

0x444D replied:
Without this, the lan led does not work and cannot be controlled. 

Inspiration to use this pinmux is from [here](https://github.com/openwrt/openwrt/blob/ed364cd4b03b289a7bb4b32c350c46eddeb6b6b9/target/linux/ath79/dts/qca9533_mikrotik_routerboard-mapl-2nd.dts#L60), similar device, leds on the same gpio outputs.

f00b4r0 replied:
OK. I'm the author of the mapl commit you mention, so I'm familiar with this code :)

I have a wap-2nd lying around somewhere, I will test your PR next week.

Meanwhile the alphabetical order is still wrong ("wap-2" comes before "wap-g"), and you might also want to add a note in the commit message about the network setup (single port device, switch bypassed) as was done in eb38af788

f00b4r0 reviewed (changes requested):
Thanks for your submission.
There are a few minor things that need polishing, see comments.

0x444D committed:
ath79: add support for MikroTik RouterBOARD wAP-2nD (wAP)

The MikroTik RouterBOARD wAP-2nd (sold as wAP) is a small
2.4 GHz 802.11b/g/n PoE-capable AP.

Specifications:
 - SoC: Qualcomm Atheros QCA9533
 - Flash: 16 MB (SPI)
 - RAM: 64 MB
 - Ethernet: 1x 10/100 Mbps (PoE in)
 - WiFi: AR9531 2T2R 2.4 GHz (SoC)
 - 3x green LEDs (1x lan, 1x wlan, 1x user)

 See https://mikrotik.com/product/RBwAP2nD for more info.

Flashing:
 TFTP boot initramfs image and then perform sysupgrade. Follow common
 MikroTik procedure as in https://openwrt.org/toh/mikrotik/common.

Note: following 781d4bfb397cdd12ee0151eb66c577f470e3377d
 The network setup avoids using the integrated switch and connects the
 single Ethernet port directly. This way, link speed (10/100 Mbps) is
 properly reported by eth0.

Signed-off-by: David Musil <0x444d@protonmail.com>

f00b4r0 commented:
Tested and working; LGTM now.
@mans0n @ynezz I think this can be merged.

ynezz commented:
Merged in e20de2244270, thanks.
